### PR TITLE
Feat(front): make public profile personal page editable

### PR DESCRIPTION
**Pull request by Skylsmoi**
https://github.com/tracim/tracim/issues/4038
https://github.com/tracim/tracim/issues/4039

## Checkpoints

**For developers**

- [ ] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [ ] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [ ] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [ ] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified

**For code reviewers**

- [ ] The code is clear enough
- [ ] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [ ] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [ ] Manual, quality tests have been done

### CODE REVIEW

File: frontend/src/component/PublicProfile/CustomFormManager.jsx [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+  }
+
+  return [
+    <DisplayTitle
+      label={props.label}
+      key={props.label}
+    />,
+    props.valueList.map((value, i) => {
+      const valueType = typeof value
+      if (Array.isArray(value)) return null // INFO - CH - 20210122 - I currently don't know if this case exists in Json schema spec
```
Comment (raphj) [inline frontend/src/component/PublicProfile/CustomFormManager.jsx:81]:
```suggestion
      if (!value || Array.isArray(value)) return null // INFO - CH - 20210122 - I currently don't know if this case exists in Json schema spec
```
To account for value === null? (typeof null === 'object')

File: frontend/src/component/PublicProfile/CustomFormManager.jsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+      const valueType = typeof value
+
+      if (Array.isArray(value)) {
+        return (
+          <DisplaySchemaArray
+            label={props.schemaObject.properties[key].title}
+            valueList={value}
+            currentKey={key}
+            schemaObject={props.schemaObject.properties[key].items}
+            key={`array_${key}_${i}`}
```
Comment (raphj) [inline frontend/src/component/PublicProfile/CustomFormManager.jsx:130]: I don't think you should put `i` in the `key` prop. `key should already be unique and we should allow React to match elements if they have the same key.
Also you don't need to use `array_` here. A unique key in the same list is sufficient.

File: frontend/src/component/PublicProfile/CustomFormManager.jsx [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+  }
+
+  handleClickToggleMode = () => {
+    const { state } = this
+    if (state.mode === MODE.VIEW) {
+      this.setState({ mode: MODE.EDIT })
+      return
+    }
+
+    if (state.mode === MODE.EDIT) {
```
Comment (raphj) [inline frontend/src/component/PublicProfile/CustomFormManager.jsx:231]:
```suggestion
    else if (state.mode === MODE.EDIT) {
```

File: frontend/src/component/PublicProfile/CustomFormManager.jsx [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+      this.setState({ mode: MODE.VIEW })
+    }
+  }
+
+  render () {
+    const { props, state } = this
+
+    return (
+      <div style={{ position: 'relative' }}>
```
Comment (raphj) [inline frontend/src/component/PublicProfile/CustomFormManager.jsx:240]: Why not have a class and put this style in stylus?

I think we should keep our usage of inline style to a minimum

File: frontend/src/component/PublicProfile/CustomFormManager.jsx (hunk 1 above)
Comment (Skylsmoi) [reply frontend/src/component/PublicProfile/CustomFormManager.jsx:81]: case handled

File: frontend/src/component/PublicProfile/CustomFormManager.jsx (hunk 4 above)
Comment (Skylsmoi) [reply frontend/src/component/PublicProfile/CustomFormManager.jsx:240]: this was wip and is fixed

File: frontend/src/component/PublicProfile/CustomFormManager.jsx (hunk 2 above)
Comment (Skylsmoi) [reply frontend/src/component/PublicProfile/CustomFormManager.jsx:130]: I have removed it from DisplaySchemaObject but I have to keep it in DisplaySchemaArray

**Comment by Skylsmoi** [comment]
closing this pr and opening it with proper account

**Pull request closed without merging**